### PR TITLE
Input driven observations ("GLM-HMM")

### PR DESCRIPTION
**Pull request by zashwood**
Hi Scott,

I hope this message finds you well.  Here is the "InputDrivenObservations" class that I wrote in order to perform the analyses for our [recent paper](https://www.biorxiv.org/content/10.1101/2020.10.19.346353v1.full.pdf).  I also wrote a notebook ("2b Input Driven Observations ("GLM-HMM")") to showcase a recovery analysis using this class when the inputs and GLM weights are in the parameter regimes described in our paper.  I checked too that my code returns no error messages when calling the `test_sample(T=10, K=4, D=1, M=2)` function within your [tests/test_basics.py](https://github.com/lindermanlab/ssm/blob/master/tests/test_basics.py) script; however, I have not updated the list of observation classes within that test script to include the InputDrivenObservations class, as this class is currently written for a one-dimensional output (`D=1`; it throws error messages if you try to call it for D>1).  I wrote it for `D=1` because this was what we needed for our paper, but would be happy to work on extending it to `D>1` if this is something that you think would be useful (in that case, I think it would also be important to allow users to specify more than one value for `C`, the number of distinct classes for each dimension of output).

Anyway, thanks again to you and your lab for writing this beautiful package!  I have learned a lot by working with it, and am looking forward to the code review process for this pull request.

Best,

Zoe

### CODE REVIEW

File: ssm/hmm.py [hunk 1]
```diff
@@ -236,7 +240,9 @@ def sample(self, T, prefix=None, input=None, tag=None, with_noise=True):
         for t in range(pad, pad+T):
             Pt = self.transitions.transition_matrices(data[t-1:t+1], input[t-1:t+1], mask=mask[t-1:t+1], tag=tag)[0]
             z[t] = npr.choice(self.K, p=Pt[z[t-1]])
-            data[t] = self.observations.sample_x(z[t], data[:t], input=input[t], tag=tag, with_noise=with_noise)
+            # TODO - revert this back to how it was
```
Comment (slinderman) [inline ssm/hmm.py:243]: looks like we don't need this comment anymore?

**Review by slinderman — COMMENTED** [review]
Looks good! I made one small note re a comment in hmm.py, but otherwise fine!

**Comment by slinderman** [comment]
Thanks @zashwood! This is really impressive work. The GLM-HMM notebook is great—I like all the analyses you've done. If you could remove that one todo comment I'll merge it into master!

File: ssm/hmm.py (hunk 1 above)
Comment (zashwood) [reply ssm/hmm.py]: Thanks Scott for catching this!  I just removed this line (and also took the opportunity to update a couple of lines of the plotting code within the jupyter notebook) and created a new commit.  Thanks very much for reviewing all of this!  I really appreciate it.

**Comment by zashwood** [comment]
Thanks very much for reviewing this Scott, and for being willing to integrate this into master!  I really appreciate it!

**Pull request merged**